test/js/ez-field-edit.js

```diff
@@ -0,0 +1,86 @@
+describe('ez-field-edit', function() {
+    let element;
+    let elementWithMultipleInputs;
+
+    beforeEach(function () {
+        element = fixture('BasicTestFixture');
+        elementWithMultipleInputs = fixture('BasicTestFixtureMultipleInputs');
+    });
+
+    it('should be defined', function () {
+        assert.equal(
+            window.customElements.get('ez-field-edit'),
+            element.constructor
+        );
+    });
+
+    describe('properties', function () {
+        describe('`invalid`', function () {
+            it('should default to false', function () {
+                assert.notOk(element.invalid);
+            });
+
+            describe('set', function () {
+                it('should be reflected to an attribute', function () {
+                    element.invalid = true;
+
+                    assert.isTrue(
+                        element.hasAttribute('invalid')
+                    );
+                });
+            });
+        });
+    });
+
+    describe('validity', function () {
+        it('should get invalid', function () {
+            element.querySelector('input').value = 'invalid value';
+            element.dispatchEvent(new CustomEvent('input'));
+            assert.isTrue(
+                element.invalid
+            );
+        });
+
+        it('should get valid', function () {
+            element.invalid = true;
+            element.querySelectorAll('input').value = 42;
+            element.dispatchEvent(new CustomEvent('input'));
+            assert.isFalse(
+                element.invalid
+            );
+        });
+    });
+
+    describe('invalid lowest common parentNode between related input and label', function () {
+        it('should set `ez-invalid` class on the lowest common parentNode', function () {
+            elementWithMultipleInputs.querySelectorAll('input')[0].value = 'invalid value';
```

Comment:
you should avoid that because this relies on the order in the fixture. Here you should rather use:
```js
elementWithMultipleInputs.querySelectorAll('#lowestCommonParent1 input').value = 'invalid value';
```

or given you use the `#lowestCommonParent1` element below:
```js
const ancestor = elementWithMultipleInputs.querySelector('#lowestCommonParent1');

ancestor.querySelector('input').value = 'invalid value';
```